man/systemd.service.xml

```diff
@@ -782,15 +782,13 @@
 
       <varlistentry>
         <term><varname>NonBlocking=</varname></term>
-        <listitem><para>Set the <constant>O_NONBLOCK</constant> flag
-        for all file descriptors passed via socket-based activation.
-        If true, all file descriptors >= 3 (i.e. all except stdin,
-        stdout, and stderr) will have the
-        <constant>O_NONBLOCK</constant> flag set and hence are in
-        non-blocking mode. This option is only useful in conjunction
-        with a socket unit, as described in
-        <citerefentry><refentrytitle>systemd.socket</refentrytitle><manvolnum>5</manvolnum></citerefentry>.
-        Defaults to false.</para></listitem>
+        <listitem><para>Set the <constant>O_NONBLOCK</constant> flag for all file descriptors passed via socket-based
+        activation. If true, all file descriptors >= 3 (i.e. all except stdin, stdout, stderr and those not passed via
+        socket-based activation) will have the <constant>O_NONBLOCK</constant> flag set and hence are in non-blocking
```

Comment:
"except … not" is double negation, that's just confusing. 

Moreover , the parens are placed incorrectly, after all the "those not passed via socket-based activation" part actually applies to the "all file descriptors >= 3", too.

Hence, I'd maybe rewrite this as:

"… If true, all file descriptors >= 3 (i.e. all except stdin, stdout, stderr), excluding those passed in via the file descriptor storage logic (see FileDescriptorStoreMax= for details), will have …"

That way the parens are placed correctly, and we avoid the double negation